### PR TITLE
Implemente train function

### PR DESCRIPTION
**Pull request by guissalustiano**
Eu tô tentando montar uma estrutura para usar redes pre treinadas do keras a partir [desse tutorial](https://www.tensorflow.org/tutorials/images/transfer_learning)
Implementei uma função `generate_model` que recebe uma classe de `tf.keras.applications` (que tem as principais redes), ela coloca um algumas layers de data augmentation e normaliza a imagem além de colocar um output para classificação binaria
Além da função `train` que recebe um modelo gerado e treina com o `mura dataset`

Eu to com alguns problemas ainda mas to subindo aqui caso alguem queria/possa tentar

### CODE REVIEW

File: fracture_detection/model/generate.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    freeze: Union[bool, int, float] = False,
+    preprocess_input: Optional[Callable] = None,
+    use_data_augmentation: bool = True,
+):
+    inputs = tf.keras.layers.Input(shape=img_shape)
+    if use_data_augmentation:
+        x = _data_augmentation(inputs)
+    if preprocess_input is not None:
+        x = preprocess_input(inputs)
+    x = base_model(x, training=False)
```
Comment (rodrigofill) [inline fracture_detection/model/generate.py:48]: n tinha que mudar esse trainable=False?

**Review by rodrigofill — APPROVED** [review]

File: fracture_detection/model/generate.py (hunk 1 above)
Comment (guissalustiano) [reply fracture_detection/model/generate.py:48]: É que como tinha umas layers de bash normalization que eu não podia frizar, eu achei mais facil frezar ele ai e desfrezar dps na `_freeze_model`. Fazendo isso antes do compile parece ser ok, mas vou segurar mais um pouco pra testar isso

Comment (guissalustiano) [reply fracture_detection/model/generate.py:48]: Eu testei aqui e tá treinando certinho
